test/integration/export-404-html/test/index.test.js

```diff
@@ -0,0 +1,43 @@
+/* eslint-env jest */
```

Comment:
do you think we can merge this with existing `test/integration/export`?

Reply:
> do you think we can merge this with the existing `test/integration/export`?

The current `test/integration/export` does export tests in one pass: first build and export a version with `trailingSlash` enabled, then export a version with `trailingSlash` disabled output to the same directory (in `beforeAll`). That's why the test case I added in #36827 passed.

Reply:
we can leverage the `export-override-404` integration test

- rename it to `export-404`
- 1 case of exsiting overriding 404 case
- another 2 cases with current trailing slash enabled and disabled cases

the purpose is to group the tests together for better readability and maintaining

Reply:
Done in [`de95c82` (#36910)](https://github.com/vercel/next.js/pull/36910/commits/de95c826e3541ed9a72685e7c514dd03b008f9ab)